src/cable/backend_core.cr

```diff
@@ -21,4 +25,32 @@ module Cable
     abstract def ping_redis_subscribe
     abstract def ping_redis_publish
   end
+
+  class RegistryBackend < BackendCore
+    REGISTERED_BACKENDS = {} of String => BackendCore.class
+
+    def self.register(uri_scheme : String, backend : BackendCore.class = self)
+      REGISTERED_BACKENDS[uri_scheme] = backend
+    end
+
+    @backend : BackendCore
+
+    def initialize
+      @backend = REGISTERED_BACKENDS[URI.parse(::Cable.settings.url).scheme].new
```

Comment:
This is the magic behind the automatic runtime backend selection.

TODO: I want to provide a better error than a plain-old [`KeyError`](https://crystal-lang.org/api/1.9.2/KeyError.html). I'm thinking maybe something like this:

```crystal
raise Cable::UnknownBackend.new("Cannot find a backend for URI scheme #{scheme.inspect}. Did you require the adapter for it?")
```